streams/src/main/java/org/apache/kafka/streams/state/internals/WindowStoreIteratorWrapper.java

```diff
@@ -0,0 +1,195 @@
+
+    // this is optimizing the case when underlying is already a bytes store iterator, in which we can avoid Bytes.wrap() costs
+    private static class WrappedWindowStoreBytesIterator extends WindowStoreIteratorWrapper<Bytes, byte[]> {
+        WrappedWindowStoreBytesIterator(final KeyValueIterator<Bytes, byte[]> underlying,
+                                        final StateSerdes<Bytes, byte[]> serdes,
+                                        final long windowSize) {
+            super(underlying, serdes, windowSize);
+        }
+
+        @Override
+        public WindowStoreIterator<byte[]> valuesIterator() {
+            return new WrappedWindowStoreIterator<byte[]>(bytesIterator, serdes) {
+                @Override
+                public KeyValue<Long, byte[]> next() {
+                    final KeyValue<Bytes, byte[]> next = bytesIterator.next();
+                    final long timestamp = WindowStoreUtils.timestampFromBinaryKey(next.key.get());
+                    return KeyValue.pair(timestamp, next.value);
+                }
+            };
+        }
+
+        @Override
+        public KeyValueIterator<Windowed<Bytes>, byte[]> keyValueIterator() {
+            return new WrappedKeyValueIterator<Bytes, byte[]>(bytesIterator, serdes, windowSize) {
+                @Override
+                public Windowed<Bytes> peekNextKey() {
+                    final Bytes next = bytesIterator.peekNextKey();
+                    final long timestamp = WindowStoreUtils.timestampFromBinaryKey(next.get());
+                    final Bytes key = WindowStoreUtils.bytesKeyFromBinaryKey(next.get());
+                    return new Windowed<>(key, WindowStoreUtils.timeWindowForSize(timestamp, windowSize));
+                }
+
+                @Override
+                public KeyValue<Windowed<Bytes>, byte[]> next() {
+                    if (!bytesIterator.hasNext()) {
+                        throw new NoSuchElementException();
+                    }
+
+                    final KeyValue<Bytes, byte[]> next = bytesIterator.next();
+                    final long timestamp = WindowStoreUtils.timestampFromBinaryKey(next.key.get());
+                    final Bytes key = WindowStoreUtils.bytesKeyFromBinaryKey(next.key.get());
+                    return KeyValue.pair(
+                        new Windowed<>(key, WindowStoreUtils.timeWindowForSize(timestamp, windowSize)),
+                        next.value
+                    );
+                }
+            };
+        }
+    }
+
+    static WindowStoreIteratorWrapper<Bytes, byte[]> bytesIterator(final KeyValueIterator<Bytes, byte[]> underlying,
+                                                                   final StateSerdes<Bytes, byte[]> serdes,
+                                                                   final long windowSize) {
+        return new WrappedWindowStoreBytesIterator(underlying, serdes, windowSize);
+    }
+
+
+    final KeyValueIterator<Bytes, byte[]> bytesIterator;
```

Comment:
I can make it protected, it's being used in the overridden methods above, I think IntelliJ wasn't smart enough to figure that out.